openedx/features/course_experience/utils.py

```diff
@@ -119,7 +120,7 @@ def mark_last_accessed(user, course_key, block):
     course_key = CourseKey.from_string(course_id)
     course_usage_key = modulestore().make_course_usage_key(course_key)
 
-    if visual_progress_enabled(course_key=course_key):
+    if new_course_outline_enabled(course_key=course_key):
```

Comment:
I changed the waffle flag here because this actually has nothing to do with visual progress, it only affects whether the third-level blocks are surfaced in the course outline.